### PR TITLE
sheldon: init at 0.6.6

### PR DESCRIPTION
**Pull request by seqizz**
###### Description of changes

A fast shell plugin manager, was not included in the tree yet.
I'm potato on rust builder but this looks OK-ish. Also I didn't test for darwin.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by happysalada** [comment]
Thank you!

**Comment by happysalada** [comment]
it's usually easier for updates to go through with the github source.
Is there a particular reason you went with fetchCrate instead of fetchFromGithub ?

**Comment by happysalada** [comment]
regarding darwin, the ci tells you if the build passes.
I'm happy to test for darwin if you change the platform to allow for it.

**Comment by seqizz** [comment]
> it's usually easier for updates to go through with the github source. Is there a particular reason you went with fetchCrate instead of fetchFromGithub ?

Honestly I just copied a working derivation. I'll switch to github. Will also enable for darwin.

**Comment by seqizz** [comment]
Nope, sadly there is something going on with git version, which fails because of some git-related env variables not set. I tried to set them but couldn't manage it. Supposedly it [shouldn't even happen](https://github.com/rossmacarthur/sheldon/blob/304815b2cf4570b0459219d52a1641489252e9fc/build.rs#L78) on nix build but no idea.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 177024` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sheldon</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Just tested on my local and it's failing at the libgit2-sys compilation step, it's actually missing the `Security` framework for darwin. Do you know how to bring that ?
- in `all-packages.nix` you have to add a line like `    inherit (darwin.apple_sdk.frameworks) Security;` (just copy and paste one of the other examples.
- you can now pass the `Security` arguments inside your derivation and add it to 
`buildInputs = lib.optional stdenv.isDarwin [ Security ];`

let me know if anything doesn't make sense. Happy to elaborate and test again.

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 177024` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sheldon</li>
  </ul>
</details>

**Comment by happysalada** [comment]
very weird, this is the failure that I got 
```
ld: library not found for -lcurl
```

**Comment by seqizz** [comment]
ofborg seems to be outputting similar output [for a while](https://gist.github.com/GrahamcOfBorg). A recently-merged PR broke master branch maybe? :eyes:

**Comment by happysalada** [comment]
ok I've rerun the build on local, and can confirm that adding `curl` to nativeBuildInputs for darwin fixes the build.

The ci issue is unrelated, and like you mentioned probably due to some borg thing right now.

**Comment by seqizz** [comment]
> ok I've rerun the build on local, and can confirm that adding `curl` to nativeBuildInputs for darwin fixes the build.

Does adding to buildInputs not suffice? Just asking to be sure since I couldn't grep any other package which adds curl as nativebuildinput.

**Comment by happysalada** [comment]
yes, mispoke, buildInputs is the right choice.

File: pkgs/tools/misc/sheldon/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      --bash <($out/bin/sheldon completions --shell bash) \
+      --zsh <($out/bin/sheldon completions --shell zsh)
+  '';
+
+  meta = with lib; {
+    description = "A fast and configurable shell plugin manager";
+    homepage = "https://github.com/rossmacarthur/sheldon";
+    license = with licenses; [ mit ];
+    maintainers = with maintainers; [ seqizz ];
+    platforms = with platforms; (linux ++ darwin);
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/sheldon/default.nix:58]:
```suggestion
```
that should be the default IIRC

Comment (seqizz) [reply pkgs/tools/misc/sheldon/default.nix]: Pushed

**Comment by happysalada** [comment]
Result of `nixpkgs-review pr 177024` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>sheldon</li>
  </ul>
</details>

**Comment by happysalada** [comment]
Darwin is unhappy, and for some weird reason it doesn't work anymore on my machine even adding curl to nativeBuildInputs.
Happy to merge when you've removed darwin from the list of supported platforms.

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by seqizz** [comment]
Alright, re-pushed without darwin security thing and only adding linux as platform.

**Comment by happysalada** [comment]
Thank you!

File: pkgs/tools/misc/sheldon/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{ lib
+, stdenv
+, fetchCrate
+, rustPlatform
+, pkg-config
+, openssl
+, installShellFiles
+, Security
```
Comment (Artturin) [inline pkgs/tools/misc/sheldon/default.nix:8]: `error: Function called without required argument "Security" at ...`

fix in https://github.com/NixOS/nixpkgs/pull/200123

**Pull request merged**